scripts/api.js

```diff
@@ -14,7 +14,14 @@ function ajax(method, apiUrl, queryParams = {}, data = {}, fetchPromise = fetch)
     let searchParams = new URLSearchParams();
     searchParams.append("url", apiUrl);
     for (const key in queryParams) {
-        searchParams.append(key, queryParams[key]);
+        const value = queryParams[key];
+        if (Array.isArray(value)) {
+            for (const item of value) {
+                searchParams.append(`${key}[]`, item);
```

Comment:
Adding `[]` seems odd as any key can be repeated in a query and the PHP code has to handle this inconsistency anyway (a single key will be a string but multiples are an array). This also increases the length of URL as we have to encode `[]`.

Maybe it would make sense to have a convention of using plural nouns (e.g. `keys` rather than `key`) when we expect some repeated field?

It's not a huge issue though.

Reply:
These are good points.

The project search API was one of the first one we implemented and the only one currently using this multi-key design. It's been in the code for over 4 years at this point and used by several external callers so we can't make breaking changes in the endpoint (using comma-separated names for keys, etc).

That doesn't mean we can't make non-breaking changes though (adding a new `fields` entry, making a new `v2` endpoint, etc) and use that instead.

I think this PR is the right fix for this immediate problem -- updating our internal ajax code to support the endpoint as it exists. I'm open to other/better ways of handling this in the API but we need to do so in ways that isn't going to break backwards compatibility.

Reply:
@cpeel your answer makes a lot of sense in the context of keeping our API backwards compatible.

I was confused with how `field[]` and `fields` would be an issue as they are both new parameters. Digging into our API, I think the concern is about parameters that are _already plurals_. I could only find `languages` but that's enough as the plural-rule-means-array would change how it works, so this is not really an option. Sorry for the noise...

Reply:
It wasn't noise and there is no apology needed -- please continue asking questions and poking at code!